src/components/dialogs/network-modifications/vsc/modification/vsc-modification-dialog.tsx

```diff
@@ -0,0 +1,274 @@
+            hvdcLineTab[MAXIMUM_ACTIVE_POWER],
+            hvdcLineTab[OPERATOR_ACTIVE_POWER_LIMIT_SIDE1],
+            hvdcLineTab[OPERATOR_ACTIVE_POWER_LIMIT_SIDE2],
+            hvdcLineTab[CONVERTERS_MODE],
+            hvdcLineTab[ACTIVE_POWER],
+            hvdcLineTab[ANGLE_DROOP_ACTIVE_POWER_CONTROL],
+            hvdcLineTab[P0],
+            hvdcLineTab[DROOP],
+            converterStation1,
+            converterStation2,
+            !!editData,
+            editData?.uuid ?? null
+        ).catch((error) => {
+            snackError({
+                messageTxt: error.message,
+                headerId: 'VscModificationError',
+            });
+        });
+    };
+
+    return (
+        <FormProvider
+            {...{
+                validationSchema: formSchema,
+                removeOptional: true,
+                ...formMethods,
+            }}
+        >
+            <ModificationDialog
+                fullWidth
+                onClear={setValuesAndEmptyOthers}
+                onSave={onSubmit}
+                aria-labelledby="dialog-modify-vsc"
+                maxWidth={'md'}
+                titleId="ModifyVsc"
+                PaperProps={{
+                    sx: {
+                        height: '95vh', // we want the dialog height to be fixed even when switching tabs
+                    },
+                }}
+                open={open}
+                keepMounted={true}
+                showNodeNotBuiltWarning={equipementId != null}
+                isDataFetching={
+                    isUpdate &&
+                    (editDataFetchStatus === FetchStatus.RUNNING ||
+                        dataFetchStatus === FetchStatus.RUNNING)
+                }
+                {...dialogProps}
+            >
+                {equipementId === null && (
+                    <EquipmentIdSelector
+                        studyUuid={studyUuid}
+                        currentNode={currentNode}
+                        defaultValue={equipementId}
+                        setSelectedId={setEquipementId}
+                        equipmentType={EQUIPMENT_TYPES.HVDC_LINE}
+                        fillerHeight={17}
+                    />
+                )}
+                {equipementId !== null && (
+                    <VscModificationForm
+                        tabIndex={tabIndex}
+                        studyUuid={studyUuid}
+                        currentNode={currentNode}
+                        equipmentId={equipementId}
+                        setTabIndex={setTabIndex}
+                        vscInfos={vscInfos}
+                        tabIndexesWithError={[]}
+                    ></VscModificationForm>
+                )}
+            </ModificationDialog>
+        </FormProvider>
+    );
+};
+
+export default VscModificationDialog;
+function snackError(arg0: { messageTxt: any; headerId: string }) {
+    throw new Error('Function not implemented.\n' + arg0.messageTxt);
+}
```

Comment:
I would prefer this function to be above the VscModificationDialog one

Reply:
done